staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -196,10 +199,11 @@ var _ ObjectTracker = &tracker{}
 // of objects for the fake clientset. Mostly useful for unit tests.
 func NewObjectTracker(scheme ObjectScheme, decoder runtime.Decoder) ObjectTracker {
 	return &tracker{
-		scheme:   scheme,
-		decoder:  decoder,
-		objects:  make(map[schema.GroupVersionResource][]runtime.Object),
-		watchers: make(map[schema.GroupVersionResource]map[string][]*watch.RaceFreeFakeWatcher),
+		scheme:    scheme,
+		decoder:   decoder,
+		versioner: objectVersioner{},
+		objects:   make(map[schema.GroupVersionResource][]runtime.Object),
+		watchers:  make(map[schema.GroupVersionResource]map[string][]*watch.RaceFreeFakeWatcher),
```

Comment:
rv doesnt increase per object, a global seq tracker here makes more sense

Reply:
According to [doc](https://github.com/kubernetes/community/blob/eb468f49f545440e8f64e4c1baedb4597a090138/contributors/devel/api-conventions.md#concurrency-control-and-consistency).  Current global sequence is side-effect of current implementation by using etcd's modifiedIndex. 
> The resourceVersion is currently backed by etcd's modifiedIndex. However, it's important to note that the application should not rely on the implementation details of the versioning system maintained by Kubernetes. We may change the implementation of resourceVersion in the future, such as to change it to a timestamp or per-object counter.

So I think per-object counter is enough.

Reply:
> So I think per-object counter is enough.

mixed feeling about that. it depends on whether we will let the fake stay on a same basis of how actual rv works for now. but its a mocking client, isnt? additionally, more stuff to consider in fake rv:

- is it filling list meta w/ a valid rv?
- does the fake get/list/watch caller receives and handles requesting rv?
- how will a per-object rv works w/ cohibitating/multi-versioning resources? (tho its not supported for now, but we might want it in the future)

Reply:
>resourceVersion: a string that identifies the internal version of this object that can be used by clients to determine when objects have changed. This value MUST be treated as opaque by clients and passed unmodified back to the server. Clients should not assume that the resource version has meaning across namespaces, different kinds of resources, or different servers. (See concurrency control, below, for more details.)
@yue9944882 the above i read from apiconventions . Are you saying that rv is a global counter currently ? My understanding is it changes per object.

Reply:
I'll researching related code and will reply later.

Reply:
> rv is a global counter currently

hey @krmayankk, exactly. or rather, in the server, rv is tracked per-storage w/ an optional compactor. while that sequence number is forced by etcd to be global

Reply:
It's already possible to store different resources in different etcd's by configuring the `--etcd-servers-overrides` flag of the apiserver, so clients can't assume a global counter, even with today's implementation.